extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/services/ContractAgreementDataFetcher.java

```diff
@@ -76,10 +79,44 @@ public List<ContractAgreementData> getContractAgreements(DSLContext dsl) {
             .toList();
     }
 
-    private @NotNull Map<String, SovityContractTerminationRecord> fetchTerminations(DSLContext dsl, List<ContractAgreement> agreements) {
+    @NotNull
+    public ContractAgreementData getContractAgreement(DSLContext dsl, String contractAgreementId) {
 
-        var agreementIds = agreements.stream().map(ContractAgreement::getId).toList();
+        val agreement = getAllContractAgreement(contractAgreementId);
+
+        val negotiationQuery = QuerySpec.max();
+        val negotiation = contractNegotiationStore.queryNegotiations(negotiationQuery)
+            .filter(it -> it.getContractAgreement().getId().equals(contractAgreementId))
+            .findFirst()
+            .orElseThrow(
+                () -> new IllegalStateException("Can't find any negotiation for contract agreement id %s".formatted(contractAgreementId)));
+
+        val transfers = getAllTransferProcesses().stream().collect(groupingBy(it -> it.getDataRequest().getContractId()));
+
+        val terminations = fetchTerminations(dsl, agreement.getId());
+
+        val maybeAsset = assetIndex.findById(agreement.getAssetId());
+        val isConsumer = negotiation.getType() == ContractNegotiation.Type.CONSUMER;
+        val asset = (maybeAsset == null || isConsumer) ? dummyAsset(agreement.getAssetId()) : maybeAsset;
+
```

Comment:
comment: this looks similar to the `getAsset()` method below. Is it viable to upgrade `getAsset()` to do it this way or is it out of scope?

Reply:
It's the same logic.

I found a way to merge the 2.